[PFBF2-0003] Edit Profile Have no Validation

SetyaG opened the pull request:
Closes #47 
***
## Description
Added validation and error label on Edit Profile page

## Related Links
List related PRs against other branches:

List | link
------ | ------
User Story | [link]()
Figma | [link]()
Issue | [link](#47)

## Todos
- [X] UI
- [ ] Coordinator
- [ ] Network
- [ ] Tests
- [X] Documentation

## Screenshot / Video
Screenshot consisting of changes made on this branch.
### UI Changes
<img width="296" alt="Screen Shot 2020-08-26 at 00 43 13" src="https://user-images.githubusercontent.com/45433184/91203667-d4575e80-e735-11ea-858d-d42d090a07a0.png">
<img width="296" alt="Screen Shot 2020-08-26 at 00 43 18" src="https://user-images.githubusercontent.com/45433184/91203696-d8837c00-e735-11ea-8744-8016c2db242c.png">
<img width="296" alt="Screen Shot 2020-08-26 at 00 43 30" src="https://user-images.githubusercontent.com/45433184/91203706-dae5d600-e735-11ea-89bf-28252ff5e488.png">
### Network Status
Network related changes will be shown here (screenshot of request status)

### Test Coverage
Test coverage result for designated changes will be shown here (screenshot of coverage)

## Notes
Additional notes that you have either technical debt, blocker, etc.

SetyaG committed:
Added logic and validation in Edit Profile page
Added error label on Edit Profile page
Provided the ability to hide keyboard on tap anywhere on the view, like in register page

Added a new constant in ProfileConstant

SetyaG committed:
Fixed bugs where the error label gets hidden when the email is correct, but the phone number is wrong

inxudianz commented on MainApp/MainApp/Feature/Profile/EditProfile/EditProfileController.swift:
```diff
@@ -15,15 +15,53 @@ class EditProfileController: MasterViewController, EditProfileProtocol {
     // MARK: - Handler
     @IBAction func saveButtonTapped(_ sender: Any) {
         viewModel?.updateProfile(name: name, phone: phone, email: email)
+        viewModel?.navigateToProfile()
```
this function is called in viewmodel after updateProfile

inxudianz reviewed (changes requested):
please also update Unit testing based on added function

SetyaG committed:
Moved Edit Profile's navigateToProfile to View Model

SetyaG committed:
Added scenario for Edit Profile unit testing
Changed naming on EditProfileController from handleField to setupField, due to same name in View Model

inxudianz committed:
Merge branch 'develop' into bugfix/PFBF2-0003
